src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -271,6 +274,7 @@ def load_arguments(self, _):
         c.argument('network_plugin', arg_type=get_enum_type(network_plugins))
         c.argument('network_plugin_mode', arg_type=get_enum_type(network_plugin_modes))
         c.argument('network_policy')
+        c.argument('network_dataplane', arg_type=get_enum_type(network_dataplanes))
```

Comment:
If you don't mind, please set this new parameter many lines below as there gathers the options for aks-preview only.

Also please update the parameter `enable_cilium_dataplane` to set it as deprecated.

`c.argument('enable_cilium_dataplane', action='store_true', is_preview=True, deprecate_info=c.deprecate(target='--enable-cilium-dataplane', redirect='--network-dataplane', hide=True))`

Reply:
Updated to add `deprecate_info`

The `--network-dataplane` flag is going to be added to the stable CLI as well. Do you still want me to move it?

Reply:
Draft PR for the stable CLI: https://github.com/Azure/azure-cli/pull/25951